scala/spark-wordcount/src/main/scala/com/example/bigtable/spark/wordcount/WordCount.scala

```diff
@@ -127,15 +127,17 @@ object WordCount {
     job.setOutputFormatClass(classOf[TableOutputFormat[ImmutableBytesWritable]])
     conf = job.getConfiguration()
 
-    val wordCounts = sc.textFile(fileName).
-      flatMap(_.split(" ")).
-      filter(_ != "").map((_, 1)).
-      reduceByKey((a, b) => a + b)
-      .map({
-        case (word, count) =>
-          (null, new Put(Bytes.toBytes(word))
-            .addColumn(ColumnFamilyBytes, ColumnNameBytes, Bytes.toBytes(count)))
-      })
+    val wordCounts = sc
+      .textFile(fileName)
+      .flatMap(_.split("\\W+"))
+      .filter(!_.isEmpty)
```

Comment:
I noticed this changed slightly, including this filter. Is this why the word count changed in the test?

Separately, I see some things changed from lambdas to `_`, like line 135. Which way (if any) is preferred?

Reply:
For `\\W+` see the comment above.

I changed it to `_` (underscores) to match the other lines (although I used a different approach in `case {...}`).

I'm OK to change it if needed. That does not show much of value regarding Spark, Dataproc and Bigtable.

Reply:
Sounds good. I was wondering which would be clearest to the user reading the sample - will defer to you on which is the preferred way.